### DISCUSSION TITLE
Bundle update output order

### DISCUSSION THREAD

**Discussion by collimarco** [ruby/rubygems#4901] [General]
Some weeks ago the output of `bundle update` has changed.

Instead of having `Fetching gem X` and then immediately `Installing gem X`, we now have the output in a confused order... 

**Example (difficult to understand):**
```
Fetching gem X
Fetching gem Y
Installing gem X
Fetching gem Z
Installing gem Y
Installing gem Z
```

Is there any way to get the output in the correct order, which is extremely more readable?

**Example (much more readable):**
```
Fetching gem X
Installing gem X
Fetching gem Y
Installing gem Y
Fetching gem Z
Installing gem Z
```

**Comment by simi** [comment]
If I remember well, this is caused by parallel installation. If you prefer slower, but properly "ordered" output, you can limit to only 1 job.

```
       --jobs=[<number>], -j[<number>]
              The maximum number of parallel download and install jobs. The default is 1.
```

**Reply by collimarco** [reply]
Isn't `-j 1` the default value? Note that I haven't passed any arguments to `bundle update`

**Reply by MSP-Greg** [reply]
I think that `1` is the default setting if `-j` is used without a number.  Otherwise, `Bundler::Settings#processor_count` (which is `Etc.nprocessors`) is used?

**Reply by collimarco** [reply]
So if I don't pass `-j` the default should be 1... however the order of the output is in random order. What should I do?

**Reply by deivid-rodriguez** [reply]
No, the default value is only `1` on Windows, because of [some issues](https://github.com/rubygems/rubygems/issues/3719) we found with parallel installation. On other platforms, `bundle install` runs at "full speed" by default. We have an issue to improve this: https://github.com/rubygems/rubygems/issues/3305.

**Reply by deivid-rodriguez** [reply]
I just checked the Windows situation. We did keep the `1` default for a while to prevent those issues I mentioned, but we have been running CI with parallel installation by default for months and it seems all good, so the default will be flipped to the number of processors on Windows too in `bundler 2.3.0`.

**Reply by collimarco** [reply]
Is there any way/option to have only **1 line of output for each gem**? Or at least to have fetching and installing in **two consecutive lines**?

**Reply by deivid-rodriguez** [reply]
Your suggestion is duly noted. In the meantime, feel free to configure `jobs` to be `1` so you can get an ordered output.

**Comment by deivid-rodriguez** [comment]
Closing in favor of #3305.
